### PR TITLE
Meta-annotate @SpringBatchTest with @ExtendWith(SpringExtension.class)

### PR DESCRIPTION
**Pull request by benas**
This PR meta-annotates `@SpringBatchTest` with `@ExtendWith(SpringExtension.class)` so it can be used in a JUnit Jupiter test without having to extend the test class with `SpringExtension`.

Issue #3647

### CODE REVIEW

**Comment by benas** [comment]
@sbrannen Hi Sam,

Could you please take a look at this PR? Let me know if it needs any updates.

Many thanks upfront!

Best regards,
Mahmoud

File: spring-batch-test/src/main/java/org/springframework/batch/test/context/SpringBatchTest.java [hunk 1]
```diff
@@ -78,7 +81,40 @@
  *
  * }
  * </pre>
+ * 
+ * For JUnit 5, this annotation can be used without having to extend the test class
+ * with {@link SpringExtension} as it is meta-annotated with {@code @ExtendWith(SpringExtension.class)}:
```
Comment (sbrannen) [inline spring-batch-test/src/main/java/org/springframework/batch/test/context/SpringBatchTest.java:86]: For the sake of clarity, I would suggest rewording this as follows.

```suggestion
 * For JUnit 5, this annotation can be used without having to manually register the
 * {@link SpringExtension} since {@code @SpringBatchTest} is meta-annotated with
 * {@code @ExtendWith(SpringExtension.class)}:
```

File: spring-batch-test/src/test/java/org/springframework/batch/test/SpringBatchTestJUnit4Tests.java [hunk 2]
```diff
@@ -46,14 +46,14 @@
...
- * Test cases for usage of {@link SpringBatchTest} annotation.
+ * Test cases for usage of {@link SpringBatchTest} annotation with JUnit 4.
  *
  * @author Mahmoud Ben Hassine
  */
 @RunWith(SpringRunner.class)
 @SpringBatchTest
-@ContextConfiguration(classes = SpringBatchTestTests.JobConfiguration.class)
-public class SpringBatchTestTests {
+@ContextConfiguration(classes = SpringBatchTestJUnit4Tests.JobConfiguration.class)
```
Comment (sbrannen) [inline spring-batch-test/src/test/java/org/springframework/batch/test/SpringBatchTestJUnit4Tests.java:55]: Just as a side note, there is no reason to explicitly configure `JobConfiguration` here since `spring-test` will automatically find any static nested `@Configuration` class by default.

```suggestion
@ContextConfiguration
```

File: spring-batch-test/src/test/java/org/springframework/batch/test/SpringBatchTestJUnit5Tests.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.test.context.ContextConfiguration;
+
+/**
+ * Test cases for usage of {@link SpringBatchTest} annotation with JUnit 5.
+ *
+ * @author Mahmoud Ben Hassine
+ */
+@SpringBatchTest
+@ContextConfiguration(classes = SpringBatchTestJUnit5Tests.JobConfiguration.class)
```
Comment (sbrannen) [inline spring-batch-test/src/test/java/org/springframework/batch/test/SpringBatchTestJUnit5Tests.java:54]: ditto

**Comment by benas** [comment]
Thank you for your feedback! I pushed an update that addresses your suggestions. Let me know if anything else should be updated.

> As a sanity check, you might want to manually verify that JUnit Jupiter artifacts don't end up as mandatory dependencies in the generated POM.

I did a quick check, and the dependency is optional as expected, here is an extract from the generated pom:

```xml
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-api</artifactId>
      <version>5.6.2</version>
      <scope>compile</scope>
      <optional>true</optional>
    </dependency>
```

**Comment by sbrannen** [comment]
> Thank you for your feedback! 

You're welcome.

> Let me know if anything else should be updated.

LGTM 👍
